### PR TITLE
PC-92: @coderabbitai

### PR DESCRIPTION
**Pull request by alexanderilyin**
## Related PRs

- #220 
- #221 

## Copilot Summary

This pull request includes several significant changes to the development environment and CI/CD workflows. The most important changes are grouped by theme as follows:

### Dev Container Updates:
* Added a new Dockerfile that includes `hadolint` and updated base images, along with new labels for container metadata.
* Introduced a new `bootstrap.sh` script for setting up the development environment, including installing Allure, pre-commit hooks, and poetry plugins.
* Updated the `devcontainer.json` to include environment variables, new features, and a post-create command to run the `bootstrap.sh` script. [[1]](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R6-R13) [[2]](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686L25-R86)

### Dependency Management:
* Updated the `apt.txt` file with new package versions for dependencies.
* Added a new `apt.sh` script to install dependencies listed in `apt.txt`.

### CI/CD Enhancements:
* Enhanced the GitHub Actions workflow with new jobs for building the dev container, installing dependencies, running tests, and executing pre-commit hooks.
* Updated the `dev-container.sh` script to include poetry installation and running tests.

### Configuration Files:
* Added a new `.coderabbit.yaml` configuration file for CodeRabbit integration with specific tone instructions and review settings.
* Updated the `.flake8` configuration to increase the max line length to 120 characters.

### Miscellaneous:
* Updated the GitHub Actions setup action to include a description and use the latest version of `actions/cache`. [[1]](diffhunk://#diff-9fb8259afcb532c9556e2d41ae1ee97cf6de54b486cef93a7ef1bf39433fa530R2) [[2]](diffhunk://#diff-9fb8259afcb532c9556e2d41ae1ee97cf6de54b486cef93a7ef1bf39433fa530L19-R20)
* Added a new section in the pull request template for GitHub Copilot summaries.

These changes collectively improve the development environment setup, streamline dependency management, and enhance the CI/CD workflows for better efficiency and maintainability.

### CODE REVIEW

**Comment by alexanderilyin** [comment]
@coderabbitai review full

File: .devcontainer/apt.sh
Comment (alexanderilyin) [inline .devcontainer/apt.sh:1]: - [ ] Document how to use it.

File: .devcontainer/bootstrap.sh
Comment (alexanderilyin) [inline .devcontainer/bootstrap.sh:1]: - [ ] Move `allure` and `poetry` plugins to `Dockerfile`

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+      "language": "python",
+      "prefix": "# "
+    }
+  ],
+  "psi-header.templates": [
+    {
+      "language": "python",
+      "template": [
+        "",
+        "OpenVMP, 2023-2024",
```
Comment (openvmp) [inline .vscode/settings.json:32]:
```suggestion
        "PartCAD, <<dateformat('YYYY')>>",
```

File: README.md [hunk 2]
```diff
@@ -8,35 +8,33 @@
...
+  - [x] Windows
   - [x] Linux
-  - [x] MacOS
+  - [x] macOS
 - Collaboration on designs
-  - [x] Versioning of CAD designs using `Git` _(like it's 2024 for real)_
-      - [x] Mechanics
-      - [ ] Electronics _(in progress)_
-      - [ ] Software _(in progress)_
+  - [x] Versioning of CAD models using `Git` _(like it's 2024 for real)_
```
Comment (openvmp) [inline README.md:34]:
```suggestion
  - [x] Versioning of CAD designs using `Git` _(like it's 2024 for real)_
```

File: README.md [hunk 3]
```diff
@@ -61,7 +59,8 @@ Subscribe on [LinkedIn], [YouTube], [TikTok], [Facebook], [Instagram], [Threads]
   - Generating with LLM/GenAI
     - [x] Google AI (`Gemini`)
     - [x] OpenAI (`ChatGPT`)
-    - [x] Any model in [Ollama](https://ollama.com/) (`Llama 3.1`, `DeepSeek-Coder-V2`, `CodeGemma`, `Code Llama` etc)
+    - [x] Any model published to [Ollama](https://ollama.com/) (`Llama 3.1`, `DeepSeek-Coder-V2`, `CodeGemma`,
```
Comment (openvmp) [inline README.md:62]:
```suggestion
    - [x] Any model in [Ollama](https://ollama.com/) (`Llama 3.1`, `DeepSeek-Coder-V2`, `CodeGemma`,
```

File: docs/source/conf.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 project = "PartCAD"
 copyright = "2024, PartCAD"
-author = "Roman Kuzmenko"
+author = "PartCAD Inc."
```
Comment (openvmp) [inline docs/source/conf.py:11]:
```suggestion
author = "PartCAD"
```

File: mkdocs/docs/development.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# Development
+
+## Documentation
+
+- `./docs/`: User documentation, based on Sphinx and available at <https://partcad.readthedocs.io/>
```
Comment (openvmp) [inline mkdocs/docs/development.md:5]: Add a sphinx onliner here?

File: mkdocs/docs/development.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+**Management Commands:**
+
+- `@coderabbitai resolve` - Resolves all the CodeRabbit review comments.
+- `@coderabbitai configuration` - Displays the current CodeRabbit configuration for the repository.
+
+**Help:**
+
+- `@coderabbitai help` - Displays available commands and usage information.
+
+[1]: https://python-poetry.org/
```
Comment (openvmp) [inline mkdocs/docs/development.md:131]: See the top level README for ideas how to make it more maintainable

File: mkdocs/docs/index.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# Technonomicon
```
Comment (openvmp) [inline mkdocs/docs/index.md:1]: PartCAD?

File: mkdocs/mkdocs.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+site_name: Technonomicon
```
Comment (openvmp) [inline mkdocs/mkdocs.yml:1]: PartCAD

File: partcad-cli/pyproject.toml [hunk 9]
```diff
@@ -7,10 +7,10 @@ keywords = ["cadquery", "build123d", "cad", "design", "openscad", "step", "stl"]
 requires-python = ">=3.10"
 license = {file = "LICENSE.txt"}
 authors = [
-  {name = "Roman Kuzmenko", email = "openvmp@proton.me" }
+  {name = "PartCAD Inc.", email = "support@partcad.org" }
```
Comment (openvmp) [inline partcad-cli/pyproject.toml:10]: Not Inc

File: pyproject.toml [hunk 10]
```diff
@@ -38,8 +49,7 @@ name = "partcad-dev"
 version = "0.1.0"
 description = "Development Environment for PartCAD"
 authors = [
-  "Roman Kuzmenko <support@partcad.org>",
-  "Alexander Ilyin <alexander@ilyin.eu>"
+  "PartCAD Inc. <support@partcad.org>",
```
Comment (openvmp) [inline pyproject.toml:52]: Drop Inc

File: .vscode/settings.json [hunk 11]
```diff
@@ -1,5 +1,44 @@
...
+        "",
+        "Author: <<author>> (<<authorEmail>>)",
+        "Created: <<date>>",
+        "",
+        "Licensed under Apache License, Version 2.0.",
+        "",
+      ]
+    }
```
Comment (alexanderilyin) [inline .vscode/settings.json:40]: Example:
```
# 
# PartCAD, 2024
# 
# Author: PartCAD (support@partcad.org)
# Created: Sat Nov 30 2024
# 
# Licensed under Apache License, Version 2.0.
# 
```

**Pull request merged**